Automattic/wp-calypso
# Update Jetpack Cloud Upsells to RNA designs

#### Changes proposed in this Pull Request
Asana task:  1164141197617539-as-1201628427568968

This PR Updates the Backup, Scan, and Search Upsells in Jetpack Cloud to new RNA designs.
Design post here: p1HpG7-fg7-p2

- Also updated the prices placeholder that are used in all product cards so that the placeholder assumes the same size & height of the actual rendered prices (regardless of the styles and font-sizes used). Prior to this fix, the price placeholder width & height was hard coded at a set value.  This fix will prevent layout-shift that occurs when the placeholder doesn't assume the same height of the price element that ultimately loads in.
- Also fixes the Jetpack Search upsell loading delay. Prior to this fix, the Search UI was taking approximately 1 second to load. Now the Search UI loads very quickly, similar to both Backup and Scan UI's.

![Screenshot on 2022-03-29 at 17-05-19](https://user-images.githubusercontent.com/11078128/160707103-1c78a1e8-c7dc-4c7b-b3db-c1f7dab3eefd.png)


#### Testing instructions

Figma designs, for reference: 1YO0iEGd6N9rffM0P7h8Ws-fi-2209%3A9986

- Checkout this PR and spin up both Calypso blue (`yarn start`) and Calypso green (`yarn start-jetpack-cloud-p`) concurrently.
- Create a JN site, connect Jetpack, and select Jetpack Free. (we want to start with no products/plans).
- Visit `http://jetpack.cloud.localhost:3001/backup/:site`, where the `:site` is the site slug of the JN site.
- Verify you see the new Upsell design and that it matches the figma designs linked above.
- Verify the price loading placeholder looks good and does not cause any layout-shift.
- Verify the CTA goes directly to checkout with the product in the cart.
- Verify the upsell looks good in mobile view.
- Now verify the same things for both Scan and Search.
- For Search, notice that the loading time is no longer delayed. (Compare with production cloud.jetpack.com)
- Now visit Calypso blue (`http://calypso.localhost:3000).  Select your JN site, and Navigate to Jetpack --> Backup. Verify the upsell is unchanged and remains the same as production.
- Do the same for Jetpack --> Scan, verify the upsell is unchanged and is same as production.
- Do the same for Jetpack --> Search, verify the upsell is unchanged and is same as production (except for that the loading time has been fixed).
- Back in Calypso green, purchase Backup and verify that after purchase, the upsell no longer appears.
- Now do the same for both Scan and Search.
- And finally, just double check the price loading placeholders look and work good on all the pricing pages:
    - `http://jetpack.cloud.localhost:3001/pricing/:site`
    - `http://calypso.localhost:3000/jetpack/connect/plans/:site`
    - `http://calypso.localhost:3000/plans/:site`


#### Screencasts

https://user-images.githubusercontent.com/11078128/160898955-3cd14bc5-bd21-4346-804c-9ab941178b81.mp4


Responsiveness:

https://user-images.githubusercontent.com/11078128/161096907-ba3bd1f2-3eca-4de9-a9ec-b3ef3d46f4d4.mp4
